src/infill/SpaghettiInfill.cpp

```diff
@@ -0,0 +1,139 @@
+/** Copyright (C) 2017 Ultimaker - Released under terms of the AGPLv3 License */
+#include "SpaghettiInfill.h"
+
+namespace cura {
+
+
+void SpaghettiInfill::generateSpaghettiInfill(SliceMeshStorage& mesh)
+{
+    coord_t layer_height = mesh.getSettingInMicrons("layer_height");
+    coord_t line_width = mesh.getSettingInMicrons("infill_line_width");
+    double layer_height_mm = INT2MM(layer_height);
+    int spaghetti_max_layer_count = std::max(1, static_cast<int>(mesh.getSettingInMicrons("spaghetti_max_height") / layer_height));
+    // TODO: account for the initial layer height
+
+    coord_t filling_area_inset = mesh.getSettingInMicrons("spaghetti_inset");
+    
+    if (mesh.getSettingInAngleDegrees("spaghetti_max_infill_angle") >= 90)
+    {
+        return; // infill cannot be combined into pillars
+    }
+    coord_t connection_inset_dist = tan(mesh.getSettingInAngleRadians("spaghetti_max_infill_angle")) * layer_height; // Horizontal component of the spaghetti_max_infill_angle
+
+    std::list<SpaghettiInfill::InfillPillar> pillar_base;
+    size_t min_layer = mesh.getSettingAsCount("bottom_layers") + 1;
+    size_t max_layer = mesh.layers.size() - 1 - mesh.getSettingAsCount("top_layers");
+    for (size_t layer_idx = min_layer; layer_idx <= max_layer; layer_idx++) //Skip every few layers, but extrude more.
+    {
+        SliceLayer& layer = mesh.layers[layer_idx];
+
+        // add infill parts to pillar_base
+        for (SliceLayerPart& slice_layer_part : layer.parts)
+        {
+            std::vector<PolygonsPart> part_infill_parts = slice_layer_part.getOwnInfillArea().splitIntoParts();
+
+            // add parts to pillar_base
+            for (PolygonsPart& infill_part : part_infill_parts)
+            {
+                SpaghettiInfill::InfillPillar& pillar = addPartToPillarBase(infill_part, pillar_base, connection_inset_dist);
+                pillar.top_slice_layer_part = &slice_layer_part;
+                pillar.last_layer_added = layer_idx;
+                pillar.layer_count++;
+            }
+        }
+
+
+        // handle finished pillars
+        for (auto it = pillar_base.begin(); it != pillar_base.end();)
```

Comment:
Bad variable name: `it`. "It" is a clown from a horror series and not applicable to pillars of infill in any way.